kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/RS3WindowFlushManager.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2025 Responsive Computing, Inc.
+ *
+ * This source code is licensed under the Responsive Business Source License Agreement v1.0
+ * available at:
+ *
+ * https://www.responsive.dev/legal/responsive-bsl-10
+ *
+ * This software requires a valid Commercial License Key for production use. Trial and commercial
+ * licenses can be obtained at https://www.responsive.dev
+ */
+
+package dev.responsive.kafka.internal.db.rs3;
+
+import dev.responsive.kafka.internal.db.RemoteWriter;
+import dev.responsive.kafka.internal.db.WindowFlushManager;
+import dev.responsive.kafka.internal.db.partitioning.TablePartitioner;
+import dev.responsive.kafka.internal.db.rs3.client.LssId;
+import dev.responsive.kafka.internal.db.rs3.client.RS3Client;
+import dev.responsive.kafka.internal.db.rs3.client.WalEntry;
+import dev.responsive.kafka.internal.stores.RemoteWriteResult;
+import dev.responsive.kafka.internal.utils.WindowedKey;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+class RS3WindowFlushManager implements WindowFlushManager<Integer> {
+  private final UUID storeId;
+  private final RS3Client rs3Client;
+  private final LssId lssId;
+  private final RS3WindowTable table;
+  private final int kafkaPartition;
+  private final PssPartitioner pssPartitioner;
+  private final Map<Integer, Optional<Long>> writtenOffsets;
+
+  private long streamTime;
+  private final Map<Integer, Rs3WindowWriter> writers = new HashMap<>();
+
+  public RS3WindowFlushManager(
+      final UUID storeId,
+      final RS3Client rs3Client,
+      final LssId lssId,
+      final RS3WindowTable table,
+      final int kafkaPartition,
+      final PssPartitioner pssPartitioner,
+      final Map<Integer, Optional<Long>> writtenOffsets,
+      final long initialStreamTime
+  ) {
+    this.storeId = storeId;
+    this.rs3Client = rs3Client;
+    this.lssId = lssId;
+    this.table = table;
+    this.kafkaPartition = kafkaPartition;
+    this.pssPartitioner = pssPartitioner;
+    this.writtenOffsets = writtenOffsets;
+    this.streamTime = initialStreamTime;
+  }
+
+  @Override
+  public String tableName() {
+    return table.name();
+  }
+
+  @Override
+  public TablePartitioner<WindowedKey, Integer> partitioner() {
+    return new PssTablePartitioner<>(pssPartitioner) {
+      @Override
+      public byte[] serialize(final WindowedKey key) {
+        // TODO: Get rid of this
```

Comment:
?

Reply:
Not sure what I had in mind. I'll get rid of the TODO since the implementation seems ok to me.